torch/cuda/_utils.py

```diff
@@ -1,11 +1,397 @@
-from typing import Any
+import ctypes
+import os
+import sys
+from typing import Any, Optional, Union
 
 import torch
 
 # The _get_device_index has been moved to torch.utils._get_device_index
 from torch._utils import _get_device_index as _torch_get_device_index
 
 
+def _get_nvrtc_version(cuda_version: int) -> str:
+    # Follows same logic as LazyNVRTC.cpp getLibVersion()
+    major = cuda_version // 1000
+    minor = (cuda_version // 10) % 10
+
+    if sys.platform == "win32":
+        if major < 11 or (major == 11 and minor < 3):
+            return f"{major}{minor}"
+        elif major == 11:
+            return "112"
+        else:
+            return f"{major}0"
+    else:
+        if major < 11 or (major == 11 and minor < 3):
+            return f"{major}.{minor}"
+        elif major == 11:
+            return "11.2"
+        else:
+            return str(major)
```

Comment:
Do you mind adding a TODO to just expose this one from native code? (Can take care of this one later)